cmd/participation/join.go

```
@@ -0,0 +1,68 @@
+/*
+Copyright IBM Corp. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package main
+
+import (
+	"bytes"
+	"fmt"
+	"io/ioutil"
+	"mime/multipart"
+	"net/http"
+	"os"
+)
+
+// Joins a OSN to a new or existing channel.
+func join(osn, tlsDir, channelID, configBlockPath string) {
+	blockBytes, err := ioutil.ReadFile(configBlockPath)
+	if err != nil {
+		panic(err)
```

A general comment, why so many panics here? Why not simply return the error.

---

Refactored. Done.